crates/viewer/re_space_view_spatial/src/transformables.rs

```diff
@@ -0,0 +1,116 @@
+use ahash::HashMap;
+use once_cell::sync::OnceCell;
+
+use nohash_hasher::IntSet;
+use re_chunk_store::{
+    ChunkStore, ChunkStoreDiffKind, ChunkStoreEvent, ChunkStoreSubscriber,
+    ChunkStoreSubscriberHandle,
+};
+use re_log_types::{EntityPath, StoreId};
+use re_types::ComponentName;
+
+// ---
+
+/// Keeps track of which entities have had any `Transform3D`-related data on any timeline at any
+/// point in time.
+///
+/// This is used to optimize queries in the `TransformContext`, so that we don't unnecessarily pay
+/// for the fixed overhead of all the query layers when we know for a fact that there won't be any
+/// data there.
+/// This is a huge performance improvement in practice, especially in recordings with many entities.
+#[derive(Default)]
+pub struct Transformables {
```

Comment:
fancy choice of name. Wonder if that's what we could call these components user facing as well hum

Reply:
how about `TransformComponentTracker` for this thing?

Reply:
Sure thing